alternative fix to mypy errors

siv2r opened the pull request:
Here is an alternative fix for mypy errors without using the `#type: ignore` comment.

Changes:
- explicitly check for infinity point using `is None` instead of `is_infinity()`
- check for the infinity point before passing the variable to `compress_point()`

siv2r committed:
alternative fix to mypy errors

- explicitly check for inf point using `is None`
- check for inf point before calling `compress_point`

jonasnick commented on reference.py:
```diff
@@ -142,11 +142,11 @@ def schnorr_presig_sign(msg: bytes, seckey: bytes, aux_rand: bytes, T: bytes) ->
     T_point = cpoint(T)
     R0 = point_add(R, T_point) # elliptic curve point R0 = R + T
-    if is_infinite(R0):
+    if R0 is None: # fail if infinity point
```
```suggestion
    if R0 is None: # fail if point at infinity
```

siv2r replied:
Fixed

jonasnick reviewed (commented):
Concept ACK

siv2r committed:
fix xonlypk & plainpk mypy errors

siv2r committed:
add tests.sh file

This file checks for generates tests vectors, check for mypy errors, and
tests our API against these vectors.

siv2r commented:
Rebased. Also, added the `tests.sh` file

jonasnick reviewed (approved):
ACK 4885e36b1e06de82ab931b5eed7d2e9d320a334b
